cosmic_ray/mutating.py

```diff
@@ -57,23 +57,25 @@ def visit_mutation_site(self, node, op, num_mutations):
         mutation, the subclass instance will be asked to perform the
         mutation.
         """
-        for idx in range(num_mutations):
-            if self._count == self._target:
-                self._activation_record = {
-                    'operator': _full_module_name(op),
-                    'occurrence': self._target,
-                    'line_number': cosmic_ray.util.get_line_number(node)
-                }
-
-                old_node = node
-                node = op.mutate(old_node, idx)
-                # add lineno and col_offset for newly creted nodes
-                ast.fix_missing_locations(node)
-                self._count += 1
-                break
-            else:
-                self._count += 1
-
+        # Calculate how many mutations are left before we reach the target
+        # mutation
+        span = self._target - self._count
+
+        # If the current operator will do at least that many mutations,
+        # then let it make the mutation now.
+        if self._target >= 0 and span < num_mutations:
```

Comment:
Maybe I'm being overly sensitive, but this check for `self._target >= 0` really bugs me. As far as I can tell, it's only there to account for some tests, and it looks like the tests could be updated relatively easily to not require this fix. Am I missing something?